src/theory/ext_theory.cpp

```diff
@@ -216,122 +203,90 @@ void ExtTheory::getSubstitutedTerms(int effort,
 
 bool ExtTheory::doInferencesInternal(int effort,
                                      const std::vector<Node>& terms,
-                                     std::vector<Node>& nred,
-                                     bool isRed)
+                                     std::vector<Node>& nred)
 {
   bool addedLemma = false;
-  if (isRed)
+  std::vector<Node> sterms;
+  std::vector<std::vector<Node> > exp;
+  getSubstitutedTerms(effort, terms, sterms, exp);
+  std::map<Node, unsigned> sterm_index;
+  NodeManager* nm = nodeManager();
+  for (unsigned i = 0, size = terms.size(); i < size; i++)
   {
-    for (const Node& n : terms)
+    bool processed = false;
+    if (sterms[i] != terms[i])
```

Comment:
can you add a comment for why this can happen/what needs to be done in the code below to account for it?